modules/@apostrophecms/module/index.js

```diff
@@ -802,6 +802,17 @@ module.exports = {
         }
       },
 
+      // Modules that have REST APIs use this method
+      // to determine if a request is qualified to access
+      // it without restriction to the `publicApiProjection`
+      canAccessApi(req) {
+        if (self.options.guestApiAccess) {
```

Comment:
I decided not to make this the new default behavior because I couldn't really see a justification for breaking the security expectations of anyone who already uses guest users in a way that doesn't relate to REST APIs etc.